src/antsibull/collection_meta_lint.py

```diff
@@ -0,0 +1,207 @@
+# Author: Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Ansible Project, 2024
+
+"""
+Classes to lint collection-meta.yaml
+"""
+
+from __future__ import annotations
+
+import os
+
+import pydantic as p
+from antsibull_core import app_context
+from antsibull_core.dependency_files import parse_pieces_file
+from antsibull_core.yaml import load_yaml_file
+
+from .collection_meta import CollectionMetadata, CollectionsMetadata, RemovalInformation
+
+
+class _Validator:
```

Comment:
Have you considered moving the validation into the Pydantic model using field validators?

Reply:
7ae1e4a257988185ae366f1b3626677e812c521a

I had to use model validators, since field validators only have access to the value of a single field. Also I had to keep some validations (that require values from outside the model) here.